src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -655,10 +656,23 @@ def _save_checkpoint(
 
         # COMPUTE THE CURRENT metric
         # IF idx IS A LIST - SUM ALL THE VALUES STORED IN THE LIST'S INDICES
-        metric = validation_results_dict[self.metric_to_watch]
+        curr_tracked_metric = validation_results_dict[self.metric_to_watch]
```

Comment:
Let's do float(validation_results_dict[self.metric_to_watch])
This way both curr_tracked_metric &  all_metrics are already contain only floats. The idea is to cast it as early as possible